styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -318,7 +338,8 @@ private Response<List<WorkflowInstanceExecutionData>> instances(
         data = storage.executionData(workflowId, start, stop);
       }
     } catch (IOException e) {
-      throw new RuntimeException("Failed to get execution data of workflow " + workflowId.toKey(), e);
+      throw new RuntimeException("Failed to get execution data of workflow " + workflowId.toKey(),
+          e);
```

Comment:
```suggestion
      throw new UncheckedIOException("Failed to get execution data of workflow " + workflowId.toKey(),
          e);
```
There is already an exception created in java to wrap an `IOException` into a Runtime one: https://docs.oracle.com/javase/8/docs/api/java/io/UncheckedIOException.html

Please change this and the others

Reply:
To be done in another PR since it's unrelated to the Workflow filtering.